src/main/java/com/wire/xenon/backend/models/Qualified.java

```diff
@@ -0,0 +1,44 @@
+package com.wire.xenon.backend.models;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.validation.constraints.NotNull;
+import java.util.Objects;
+import java.util.UUID;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class Qualified {
```

Comment:
I have renamed it into `QualifiedId` , agree that having and adjective is not great.
I also like to have stronger types to avoid confusions and errors in constructors and methods but I also see some issues here:
- The types would be identical in logic, so there would be either duplicated code or a common abstract class and 2 empty classes extending that, which adds some complexity and confusion
- Any other kind of ID (messages, events) is still UUID or String, won't be coherent
- It complicates things a little bit if in the near future some other id need to migrate to fully qualified (need a new class)

In Kotlin there is `typealias` exactly for this kind of issue, but in Java we have only full classes.
I lean on keeping it simple but I see the value in your suggestion, both things seem reasonable

Reply:
Fair enough, this might be a topic for a bigger refactoring, not for the current effort of "make what's there work".